github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitChecks.java

```diff
@@ -0,0 +1,146 @@
+ * </p>
+ *
+ * @author Kanstantsin Shautsou
+ * @author Jae Gangemi
+ */
+public class GitHubBranchCommitChecks extends GitHubBranchEvent {
+    private static final String DISPLAY_NAME = "Commit Checks";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubBranchCommitChecks.class);
+
+    private List<GitHubBranchCommitCheck> checks = new ArrayList<>();
+
+    /**
+     * For groovy UI
+     */
+    @Restricted(value = NoExternalUse.class)
+    public GitHubBranchCommitChecks() {
+    }
+
+    @DataBoundConstructor
+    public GitHubBranchCommitChecks(List<GitHubBranchCommitCheck> checks) {
+        this.checks = checks;
+    }
+
+    @Override
+    public GitHubBranchCause check(GitHubBranchTrigger trigger, GHBranch remoteBranch, @CheckForNull GitHubBranch localBranch,
+            GitHubBranchRepository localRepo, TaskListener listener)
+        throws IOException {
+        if (localBranch == null) {
+            LOGGER.info("First build of branch [%s] detected", remoteBranch.getName());
+            return null;
+        }
+
+        GHCompare.Commit[] commits = getComparedCommits(localBranch, remoteBranch);
+        List<GitHubBranchCause> causes = checks.stream()
+                .map(event -> event.check(remoteBranch, localRepo, commits))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        String name = remoteBranch.getName();
+        if (!causes.isEmpty()) {
+            GitHubBranchCause cause = findFilterCause(causes);
+            if (cause != null) {
+                LOGGER.info("Commits for branch [{}] filtered: {}", name, cause.getReason());
+            } else {
+                cause = causes.get(0);
+                LOGGER.info("Commits for branch [{}] allowed: {}", name, cause.getReason());
+            }
+            return cause;
+        }
+
+        return null;
+    }
+
+    // visible for testing to avoid complex mocking
+    GHCompare.Commit[] getComparedCommits(GitHubBranch localBranch, GHBranch remoteBranch) throws IOException {
+        String previous = localBranch.getCommitSha();
+        String current = remoteBranch.getSHA1();
+
+        LOGGER.debug("Comparing previous hash [{}] with current hash [{}]", previous, current);
+        return remoteBranch.getOwner()
+                .getCompare(previous, current)
+                .getCommits();
+    }
+
+    public List<GitHubBranchCommitCheck> getChecks() {
+        return checks;
+    }
+
+    void setChecks(List<GitHubBranchCommitCheck> checks) {
+        this.checks = checks;
+    }
+
+    private GitHubBranchCause findFilterCause(List<GitHubBranchCause> causes) {
+        GitHubBranchCause cause = causes.stream()
+                .filter(GitHubBranchCause::isSkip)
+                .findFirst()
+                .orElse(null);
+
+        if (cause == null) {
+            return null;
+        }
+
+        LOGGER.debug("Cause indicated build should be skipped: {}", cause.getReason());
```

Comment:
How it indicated? Skipp happens ONLY when cause object exists and cause.isSkip() is true